frontend/src/components/c-ramp.vue

```diff
@@ -39,7 +39,44 @@ import brokenLinkDisabler from '../mixin/brokenLinkMixin.ts';
 export default {
   mixins: [brokenLinkDisabler],
   name: 'c-ramp',
-  props: ['groupby', 'user', 'tframe', 'avgsize', 'sdate', 'udate', 'mergegroup', 'fromramp', 'filtersearch'],
+  props: {
+    groupby: {
+      type: String,
+      default: 'groupByRepos',
+    },
+    user: {
+      type: Object,
```

Comment:
I think it will be more helpful to define a type for the places that you need to put `Object`. This will help a lot with the typescript migration in future as well. You can perhaps define the types in a separate util file in `frontend`.

Reply:
Yes, that sounds better. I have added classes for those objects with pre-defined attributes to have better type validation.